[Documentation] [stdlib] [task] - Review Module.md file after redesign (#14544)
ballerina-platform/ballerina-lang: stdlib/task/src/main/ballerina/task/Module.md

## Module overview

Task Listeners and Task Schedulers can be used to perform tasks periodically. This module provides the functionality to configure and manage Task Listeners and Task Schedulers.

### Task Listeners

A Task `Listener` can be used to create a service listener, which will be triggered at specified times.

> Below are the two types of configurations that an be used to configure a Task Listener, either as a timer or as an appointment.

Reviewer: ```suggestion
Below are the two types of configurations that can be used to configure a Task Listener, either as a timer or as an appointment.
```

## Module overview

> This module provides functionality to configure and manage Task Listeners and Task Schedulers, which will be executed periodically.

Reviewer 2: "which will be executed periodically." <- What is meant by this? What does it mean to execute a task listener or a task scheduler periodically?

> This module provides functionality to configure and manage Task Listeners and Task Schedulers, which will be executed periodically.

Reviewer 3: ```suggestion
This module provides the functionality to configure and manage Task Listeners and Task Schedulers, which will be executed periodically.
```

### Task Listeners

Task `Listener` can be used to create a service listener, which will trigger on specified times. Listener can be configured using listener configurations.
There are two types of configurations for a Listener.
- `TimerConfiguration`
- `AppointmentConfiguration`

If a task is needed to be run periodically, `TimerConfiguration` can be used. `TimerConfiguration` consists of three fields, from which two are optional.

> - `interval` - Timer interval by which the listener should trigger. This should be given in Milliseconds.

Reviewer 2: ```suggestion
- `interval` - Timer interval by which the listener should trigger. This should be given in milliseconds.
```

> - `initialDelay` - [Optional] Initial delay before the task should trigger. If this is set to `0`, task will run immediately. If the field is not set, interval will be taken as the initial delay. This should be given in Milliseconds.

Reviewer 2: ```suggestion
- `initialDelay` - [Optional] Initial delay before the task should trigger. If this is set to `0`, task will run immediately. If the field is not set, interval will be taken as the initial delay. This should be given in milliseconds.
```

> - `noOfRecurrences` - [Optional] If there's a requirement to run a particular task only for a number of times, this field can be used. This should be given as an int.

Reviewer 2: ```suggestion
- `noOfRecurrences` - [Optional] If there is a requirement to run a particular task only for a number of times, this field can be used. This should be given as an `int`.
```
Avoid using shortened forms of phrases such as "there's"
